src/lib.rs

```diff
@@ -144,21 +148,224 @@ impl<T: MemoryAddress> AddrRange<T> {
 }
 
 /// An iterator over a memory range
-pub struct AddrIter<T: MemoryAddress> {
+#[allow(private_bounds)]
+pub struct AddrIter<T: MemoryAddress, I: IterInclusivity = NonInclusive> {
     current: T,
-    end: T,
+    end: Option<T>, // None here indicates that this is exhausted
+    _phantom: PhantomData<I>,
 }
-impl<T: MemoryAddress> Iterator for AddrIter<T> {
+
+trait IterInclusivity: 'static {
+    fn exhausted<T: Ord>(start: &T, end: &T) -> bool;
+}
+pub enum NonInclusive {}
```

Comment:
Minor nitpicks: Could you please add a doc comment to these `pub` items? Also, the line spacing is a little inconsistent.